lib/fireblocks/api/request.rb

```diff
@@ -55,7 +55,8 @@ def request_headers(body)
       {
         'X-API-Key' => Fireblocks.configuration.api_key,
         'Authorization' => "Bearer #{token(body)}",
-        'Content-Type' => 'application/json'
+        'Content-Type' => 'application/json',
+        'Idempotency-Key' => SecureRandom.uuid
```

Comment:
This is almost exactly what we need.. but we need a way of knowing/remembering this key so we can send it on subsequent requests.

Reply:
Are we planning to store the keys using Redis?

Reply:
No, maybe we should pass the key in. Maybe it should be the `onchain_transaction_id`

Reply:
In this file, we need to:

1) update `self.post` to accept `headers: {}`
2) update `post` to accept `headers: {}`
3) inside of `post` (not `self.post`), merge `request_headers` w/ `headers`